app-text/booksorg/patches/booksorg-0.3.patchset

```diff
@@ -0,0 +1,242 @@
+ #include <QString>
+ #if QT_VERSION >= 0x050000
+-#include <poppler-qt5.h>
++#include <poppler/qt5/poppler-qt5.h>
+ #else
+ #include <poppler-qt4.h>
+ #endif
+diff --git a/booksorganizer.pro b/booksorganizer.pro
+old mode 100644
+new mode 100755
+index 2162d85..6cf9cb0
+--- a/booksorganizer.pro
++++ b/booksorganizer.pro
+@@ -27,7 +27,10 @@ win32{
+     LIBS         += -L poppler-win\bin -lpoppler-qt5
+     RC_FILE = myapp.rc
+ #    QT +=winextras
+-}else{
++}
++haiku{
++	LIBS += -lpoppler -lpoppler-qt5
++	}else{
+ 
+         equals(QT_MAJOR_VERSION, 5) {
+           INCLUDEPATH  += /usr/include/poppler/qt5
+diff --git a/documentview.h b/documentview.h
+old mode 100644
+new mode 100755
+index dd911c0..7e68d96
+--- a/documentview.h
++++ b/documentview.h
+@@ -33,7 +33,8 @@ along with qpdfview.  If not, see <http://www.gnu.org/licenses/>.
+ 
+ #include <QtWidgets>
+ #include <QtPrintSupport>
+-#include <poppler-qt5.h>
++#include <poppler/qt5/poppler-qt5.h>
++
+ 
+ #else
+ #include <poppler-qt4.h>
+diff --git a/pageitem.h b/pageitem.h
+old mode 100644
+new mode 100755
+index 3606696..f969b8e
+--- a/pageitem.h
++++ b/pageitem.h
+@@ -26,8 +26,7 @@ along with qpdfview.  If not, see <http://www.gnu.org/licenses/>.
+ #include <QtGui>
+ 
+ #if QT_VERSION >= QT_VERSION_CHECK(5,0,0)
+-#include <poppler-qt5.h>
+-
++#include <poppler/qt5/poppler-qt5.h>
+ #include <QtWidgets>
+ #include <QtConcurrent>
+ #else
+diff --git a/searchthread.h b/searchthread.h
+old mode 100644
+new mode 100755
+index 5e36df5..9028025
+--- a/searchthread.h
++++ b/searchthread.h
+@@ -31,7 +31,7 @@ along with qpdfview.  If not, see <http://www.gnu.org/licenses/>.
+ #include <QStandardItemModel>
+ 
+ #if QT_VERSION >= QT_VERSION_CHECK(5,0,0)
+-#include <poppler-qt5.h>
++#include <poppler/qt5/poppler-qt5.h>
+ #else
+ #include <poppler-qt4.h>
+ #endif // QT_VERSION
+diff --git a/searchtools.cpp b/searchtools.cpp
+old mode 100644
+new mode 100755
+index 3bf60ca..c64a4a4
+--- a/searchtools.cpp
++++ b/searchtools.cpp
+@@ -24,6 +24,7 @@
+ #include "eicon.h"
+ #include <QtGui>
+ #include <QSettings>
++#include <QAction>
+ 
+ SearchTools::SearchTools(bool isPdf, QWidget *parent) :
+     QWidget(parent),
+-- 
+2.19.0
+
+
+From d3bd693f2486c105a008a422ace83189ccfb1e16 Mon Sep 17 00:00:00 2001
+From: mazbrili <mazbrili@yahoo.com>
+Date: Sun, 21 Oct 2018 06:29:07 +0700
+Subject: path fix for haiku
+
+
+diff --git a/bookinfo.cpp b/bookinfo.cpp
+index 1c69276..bfac9c1 100755
+--- a/bookinfo.cpp
++++ b/bookinfo.cpp
+@@ -36,9 +36,16 @@ BookInfo::BookInfo()
+ {
+ 
+ QSettings settings;
+-
++#if not defined(Q_OS_HAIKU)
+ QString appdir=(settings.value("BooksPath",
+                                              QDir::homePath()+"/."+QApplication::applicationName()).toString());
++#elif defined(Q_OS_HAIKU)
++QString appdir=(settings.value("BooksPath",
++											QDir::home().absolutePath() + QDir::separator() + 
++        									"config" + QDir::separator() + "settings" + QDir::separator() +
```

Comment:
Thanks @mazbrili for the updated patchset!
I did not test it yet, but if you would like to improve it a little by using the [find_direction](https://github.com/haikuports/haikuports/wiki/FindDirectory#cc) function here's a pointer to a patch that was recently done for another app:
https://github.com/haikuports/haikuports/blob/b2c69499cb536c0b1d391aea872b8c6bcbd0d004/games-strategy/colobot/patches/colobot-0.1.11.patchset#L218
BTW, @alaviss is a find_directory expert and will probably be able to guide you if you need help.

Reply:
No problem. We forgot to mention the find_directory function earlier but that's OK. We can merge the PR without it.

Reply:
Qt has [this][0] extremely convenient function that could be used instead of `find_directory()`

Here is how to get the exact path you were constructing:

```cpp
QStandardPaths::writableLocation(QStandardPaths::AppConfigLocation);
```

which is really simple, isn't it? :) And as a plus, it's portable to all other OS-es, so if the author of the program would like to have their configuration folder moved to standard paths, they could just use this with no effort involved.

[0]: http://doc.qt.io/qt-5/qstandardpaths.html#writableLocation

Reply:
oh.. i remember used that function..  hmm.... let me dig my memory

Reply:
patch update.. please check it if you have time @alaviss